checker/src/main/java/org/checkerframework/checker/determinism/DeterminismAnnotatedTypeFactory.java

```diff
@@ -269,6 +269,38 @@ public Void visitMemberSelect(
             }
             return super.visitMemberSelect(node, annotatedTypeMirror);
         }
+
+        /**
+         * Reports an error if {@code node} represents explicitly constructing a {@code @Det} or
+         * {@code @PolyDet} {@code HashSet}. If one these annotation wasn't explicitly written, but
+         * the constructor would resolve to {@code @Det}, inserts {@code @OrderNonDet} instead. If
+         * it would resolve to any variant of {@code @PolyDet}, replaces it with {@code @NonDet}
+         *
+         * @param node a tree representing instantiating a class
+         * @param annotatedTypeMirror the type to modify if it represents an invalid constructor
+         *     call
+         * @return visitNewClass() of the super class
+         */
+        @Override
+        public Void visitNewClass(NewClassTree node, AnnotatedTypeMirror annotatedTypeMirror) {
+            IdentifierTree identifier = getNewClassClassName(node);
+            if (identifier != null && identifier.getName().contentEquals("HashSet")) {
```

Comment:
It's better to not rely on String comparisons for identifying types.
You could instead use: 
```
types.isSameType(annotatedTypeMirror.getUnderlyingType(), TypesUtils.typeFromClass(HashSet.class, types, processingEnv.getElementUtils()));
```
For example, see:
https://github.com/t-rasmud/checker-framework/blob/90a589d218ba452d9a93859b5c6a3bf9932f7624/checker/src/main/java/org/checkerframework/checker/determinism/DeterminismTransfer.java#L141